cloudferrylib/os/compute/cold_evacuate.py

```diff
@@ -0,0 +1,354 @@
+
+
+@contextlib.contextmanager
+def install_ssh_keys(config, *hosts):
+    """
+    Generate and put public and private SSH keys to hosts that are listed in
+    hosts to make cold migration work.
+    :param config: CloudFerry config
+    :param hosts: list of hosts where to install keys
+    """
+    ssh_user = config.cloud.ssh_user
+    ssh_password = config.cloud.ssh_sudo_password
+    home_path = cfglib.CONF.evacuation.nova_home_path
+    nova_user = cfglib.CONF.evacuation.nova_user
+    ssh_config = '\\n'.join(['UserKnownHostsFile /dev/null',
+                             'StrictHostKeyChecking no'])
+    ssh_path = '/'.join([home_path, '.ssh'])
+    ssh_backup_base = '/'.join([home_path, '.ssh_backup'])
+
+    key = RSA.generate(2048, os.urandom)
+    public_key = key.exportKey('OpenSSH').replace('\n', '\\n')
+    private_key = key.exportKey('PEM').replace('\n', '\\n')
+
+    ssh_backups = {}
+    for host in hosts:
+        runner = remote_runner.RemoteRunner(host, ssh_user,
+                                            password=ssh_password,
+                                            sudo=True)
+        ssh_backup_path = '/'.join([ssh_backup_base,
+                                    os.urandom(8).encode('hex')])
+        try:
+            runner.run('test -e "{path}"', path=ssh_path)
+            runner.run('mkdir -p {backup_base}', backup_base=ssh_backup_base)
+            runner.run('mv "{path}" "{backup_path}"', path=ssh_path,
+                       backup_path=ssh_backup_path)
+            ssh_backups[host] = ssh_backup_path
+        except remote_runner.RemoteExecutionError:
+            LOG.debug("Dot SSH directory not found, skipping backup")
+
+        runner.run('mkdir -p "{path}"', path=ssh_path)
+        runner.run('echo -e "{key}" > "{path}"', key=public_key,
+                   path='/'.join([ssh_path, 'authorized_keys']))
+        runner.run('echo -e "{key}" > "{path}"', key=private_key,
+                   path='/'.join([ssh_path, 'id_rsa']))
+        runner.run('echo -e "{config}" > "{path}"', config=ssh_config,
+                   path='/'.join([ssh_path, 'config']))
+        runner.run('chmod 0600 "{path}"', path='/'.join([ssh_path, 'id_rsa']))
+        runner.run('chown -R "{user}:{user}" "{path}"',
+                   user=nova_user, path=ssh_path)
+    try:
+        yield
+    finally:
+        for host in hosts:
+            runner = remote_runner.RemoteRunner(host, ssh_user,
+                                                password=ssh_password,
+                                                sudo=True)
+            runner.run('rm -rf "{path}"', path=ssh_path)
+            ssh_backup_path = ssh_backups.get(host)
+            if ssh_backup_path is not None:
+                runner.run_ignoring_errors(
+                    'mv "{backup_path}" "{path}"',
+                    backup_path=ssh_backup_path, path=ssh_path)
+
+
+def change_to_pre_migration_state(compute_api, instance_id):
```

Comment:
Looks similar to `cloudferrylib.os.compute.nova_compute.NovaCompute#change_status`, please add `TODO` above to refactor later